evmcore/evm.go

```diff
@@ -51,12 +51,10 @@ func NewEVMBlockContext(header *EvmHeader, chain DummyChain, author *common.Addr
 		baseFee = new(big.Int).Set(header.BaseFee)
 	}
 
-	// Before Sonic upgrade random is always empty
-	// and difficulty is not 0
 	difficulty := big.NewInt(1)
 	if header.PrevRandao.Cmp(common.Hash{}) != 0 {
 		random = &header.PrevRandao
-		difficulty.SetUint64(0)
+		difficulty.Set(random.Big())
```

Comment:
If I read [EIP-4399](https://eips.ethereum.org/EIPS/eip-4399#using-mixhash-field-instead-of-difficulty) correctly, difficulty is supposed to be 0 after the Merge update (Paris).

Reply:
Thats right, although this set actually changes the `prevrandao` in `header` (the contract test passes with my change) and if I add this condition to the contract test, it actually passes 
```	
if block.Difficulty().Uint64() != 0 {
	t.Errorf("incorrect block difficulty: %d", block.Difficulty().Uint64())
}
```

Seems like some weird stuff happens behind the curtains...

Reply:
I believe this happens because `tosca` version in sonic develop is outdated.
Tosca has fixed this in [this](https://github.com/Fantom-foundation/Tosca/commit/d6d14c1b9515b7ae8cdda15b781405888ce84b6c) commit last week.
While we use [this](https://github.com/Fantom-foundation/Tosca/commit/aa99babe0a10a3facbaba4f67382379e568d44b3) version 3 weeks old.

Reply:
https://github.com/Fantom-foundation/Sonic/pull/273 solves this issue

Reply:
Forgot to rebase against `develop` - it already contains the necessary fix.